### PR TITLE
feat: visionOS support

### PR DESCRIPTION
**Pull request by atierian**
## Description
Adds preview support for visionOS.

**Important Note**: As with macOS, visionOS appears to require apps add the Keychain Sharing entitlement and a keychain group for keychain access.

#### Changes
- Adds `#if os(xrOS)` checks to:
  - `DeviceInfo` returns `.zero` for `screenBounds` because `UIScreen` isn't available
  - DevMenu. Also `UIScreen` related. I'd imagine we can get this working because it's only used in a single spot. Disabling it on visionOS for now.
- Background URLSessions are not available when running pure visionOS. Storage session of `os(xrOS)` is now foreground.
- useragent version is `visionos-preview`

#### Known Issues
- Some CoreML APIs have different behavior on visionOS. This only affects you if you're using the CoreMLPredictionsPlugin standalone or explicitly setting `.offline` only for requests through the AWSPredictionsPlugin.

- UI tests appear to work on visionOS with some limitations. This affects our PushNotifications integration tests because they're set up as UI test cases. Problematic test cases were manually tested to verify expected behavior.
> There is no UI for simulating Apple Vision Pro’s immersion crown. (109429267)
Workaround: Use XCTest’s XCUIDevice. rotateDigitalCrown(delta:) method. [[source](https://developer.apple.com/documentation/xctest/user_interface_tests)]

> UI testing isn’t available to apps you build using the visionOS SDK. You can still use it to test compatible iPad and iPhone apps that you build using the iOS SDK but run in visionOS. [[source](https://developer.apple.com/documentation/xcode-release-notes/xcode-15-release-notes#visionOS-Simulator)]

#### CI/CD
- Unit tests are set up to run for visionOS (in addition with the other platforms).
- Integration tests are currently run locally until we set up access for Xcode 15 beta 2 with GitHub Actions.

#### Analytics
analytics_pinpoint_integration_tests

#### API
api_functional_integration_tests
api_graphql_lambda_integration_tests
api_rest_iam_integration_tests

#### Auth
auth_integration_tests

#### DataStore
datastore_auth_cognito_integration_tests
datastore_auth_iam_integration_tests
datastore_cpk_integration_tests
datastore_lazyload_inegration_tests
datastore_transformerv2_integration_tests

#### Geo
geo_integration_tests

#### Predictions
predictions_integration_tests

#### PushNotifications
push_notifications_ui_integration_tests

#### Storage
storage_integration_tests

## General Checklist

- [ ] ~Added new tests to cover change, if needed~
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [x] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] ~New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`~
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift [hunk 1]
```diff
@@ -16,7 +16,7 @@ import Amplify
 public class AmplifyAWSServiceConfiguration {
 
     /// - Tag: AmplifyAWSServiceConfiguration.amplifyVersion
-    public static let amplifyVersion = "2.12.0"
+    public static let amplifyVersion = "visionos-preview"
```
Comment (royjit) [inline AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift:19]: no need of a number? something like `visionos-preview-0.0.1`, will be useful in keeping track of bug fixes.

**Review by royjit — APPROVED** [review]
LGTM, minor question added.

**Comment by atierian** [comment]
The watchOS test failing in CI passes locally - investigating.
```
[AWSPinpointAnalyticsPluginUnitTests.AWSPinpointAnalyticsPluginClientBehaviorTests testIdentifyUser]' passed (0.012 seconds).
Test Suite 'AWSPinpointAnalyticsPluginClientBehaviorTests' passed at 2023-06-29 12:09:30.711.
     Executed 1000 tests, with 0 failures (0 unexpected) in 6.806 (6.993) seconds
```

File: AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift (hunk 1 above)
Comment (atierian) [reply AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift]: Good call. Addressed in [e5e51ed](https://github.com/aws-amplify/amplify-swift/pull/3058/commits/e5e51eda6ef435d295c822df2bcd401238a472c1).

**Review by royjit — APPROVED** [review]

**Pull request merged**